src/test/unittest/granularity.py

```diff
@@ -89,6 +92,26 @@ def testdir_defined(cls, config):
         else:
             return True
 
+    def _check_usc_req_is_met(self, tc):
+        require_usc, _ = ctx.get_requirement(tc, 'require_usc', False)
+        if not require_usc:
+            return True
+
+        basedir = self.testdir.replace(self.testdir, '')
```

Comment:
I think basedir equals an empty string, no matter the value of testdir, which doesn't seem right.